File: careapp-backend/models/AuthorizedPerson.js
const mongoose = require('mongoose');

const authorizedPersonSchema = new mongoose.Schema({
  id_AP: { type: mongoose.Schema.Types.ObjectId, auto: true },
  name: { type: String, required: true },
  phone_number: { type: String, default: '' },
  national_id: { type: String, default: '' },
  id_U_CL: { type: mongoose.Schema.Types.ObjectId, ref: 'Client', required: true },
  email: { type: String, required: true, unique: true, lowercase: true },
  relationship: { type: String, default: '' },
  canTrack: { type: Boolean, default: true },
  canChat: { type: Boolean, default: true },
  canViewLocation: { type: Boolean, default: true },
  userId: { type: mongoose.Schema.Types.ObjectId, ref: 'User', required: true },
  createdAt: { type: Date, default: Date.now }
});

module.exports = mongoose.model('AuthorizedPerson', authorizedPersonSchema);

File: careapp_mobile/lib/models/authorized_person.dart
// lib/models/authorized_person.dart
class AuthorizedPerson {
  final String id;
  final String fullName;
  final String email;
  final String phoneNumber;
  final String relationship;
  final bool canTrack;
  final bool canChat;
  final bool canViewLocation;
  final DateTime? invitedAt;

  AuthorizedPerson({
    required this.id,
    required this.fullName,
    required this.email,
    required this.phoneNumber,
    required this.relationship,
    this.canTrack = true,
    this.canChat = true,
    this.canViewLocation = true,
    this.invitedAt,
  });

  factory AuthorizedPerson.fromJson(Map<String, dynamic> json) {
    return AuthorizedPerson(
      id: json['_id'] ?? json['id'] ?? '',
      fullName: json['fullName'] ?? json['name'] ?? '',
      email: json['email'] ?? '',
      phoneNumber: json['phoneNumber'] ?? '',
      relationship: json['relationship'] ?? '',
      canTrack: json['canTrack'] ?? true,
      canChat: json['canChat'] ?? true,
      canViewLocation: json['canViewLocation'] ?? true,
      invitedAt: json['invitedAt'] != null ? DateTime.parse(json['invitedAt']) : null,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'fullName': fullName,
      'email': email,
      'phoneNumber': phoneNumber,
      'relationship': relationship,
      'canTrack': canTrack,
      'canChat': canChat,
      'canViewLocation': canViewLocation,
    };
  }
}